Zend/Optimizer/zend_cfg.c

```diff
@@ -90,20 +99,23 @@ static void zend_mark_reachable(zend_op *opcodes, zend_cfg *cfg, zend_basic_bloc
 				if (i == b->successors_count - 1) {
 					/* Tail call optimization */
 					if (succ->flags & ZEND_BB_REACHABLE) {
-						return;
+						finished = true;
+						break;
 					}
 
 					b = succ;
 					break;
 				} else {
```

Comment:
Iterative algorithm doesn't need tail call optimization any more.
This code may be removed as well as ``while (!finished) {`` loop.